addons/edi_stock/models/edi_move_cancellation_report_document.py

```diff
@@ -0,0 +1,129 @@
+"""EDI stock move request records"""
+
+from odoo import api, fields, models
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class StockMove(models.Model):
+    """Extend ``stock.picking`` to include the EDI Stock Cancellation document"""
+
+    _inherit = "stock.move"
+
+    edi_stock_cancellation_doc_id = fields.Many2one(
+        "edi.document",
+        string="EDI Stock Cancellation Document",
+        required=False,
+        readonly=True,
+        index=True,
+    )
+
+
+class EdiDocument(models.Model):
+    """Extend ``edi.document`` to include stock move cancellation records"""
+
+    _inherit = "edi.document"
+
+    move_cancellation_ids = fields.One2many(
+        "edi.record.stock.cancellation",
+        "doc_id",
+        string="Cancelled Stock Moves",
+    )
+
+
+class EdiStockCancellationDocument(models.AbstractModel):
+    """Stock Cancellation document model
+
+    Stock Cancellation documents contain all lines cancelled since the previous
+    document.
+    """
+
+    _name = "edi.document.stock.cancellation"
+    _inherit = "edi.move.report.document"
+    _description = "Stock Cancellations"
+
+    _edi_move_report_via = "edi_stock_cancellation_doc_id"
+
+    @api.model
+    def move_report_domain(self, _doc):
+        """
+        Override to only include cancelled moves for picking types which have
+        u_enable_pick_cancellation_email_notifs enabled
+        """
+        StockPickingType = self.env["stock.picking.type"]
+        picking_types_to_consider = StockPickingType.search(
+            [("u_enable_pick_cancellation_email_notifs", "=", True)]
+        )
+        domain = [
+            ("state", "=", "cancel"),
+            ("picking_type_id", "in", picking_types_to_consider.ids),
+        ]
+        if self._edi_move_report_via is not None:
```

Comment:
just realised i do not need this, will change and just add it directly to the domain